diffmatchpatch/diff.go

```diff
@@ -830,8 +836,8 @@ func (dmp *DiffMatchPatch) DiffCleanupSemanticLossless(diffs []Diff) []Diff {
 			if commonOffset > 0 {
 				commonString := edit[len(edit)-commonOffset:]
 				equality1 = equality1[0 : len(equality1)-commonOffset]
-				edit = append(commonString, edit[:len(edit)-commonOffset]...)
-				equality2 = append(commonString, equality2...)
+				edit = append(copyRuneSlice(commonString), edit[:len(edit)-commonOffset]...)
```

Comment:
These append calls could be optimized. With the current code, we copy commonString into a new right-sized slice and then append to it, guaranteeing that it will be copied again into a new slice. You could have a `copyAppend` (`copyAndAppend`?) function that allocates a slice once to hold both arguments.